lib/core/src/sdk.rs

```diff
@@ -1132,42 +1165,79 @@ impl LiquidSdk {
     }
 
     /// Prepares to pay to a Bitcoin address via a chain swap.
-    ///
-    /// # Arguments
-    ///
-    /// * `req` - the [PreparePayOnchainRequest] containing:
-    ///     * `receiver_amount_sat` - the amount in satoshi that will be received
-    ///     * `sat_per_vbyte` - the optional fee rate of the Bitcoin claim transaction. Defaults to the swapper estimated claim fee
     pub async fn prepare_pay_onchain(
         &self,
         req: &PreparePayOnchainRequest,
     ) -> Result<PreparePayOnchainResponse, PaymentError> {
         self.ensure_is_started().await?;
 
-        let receiver_amount_sat = req.receiver_amount_sat;
+        let balance_sat = self.get_info().await?.balance_sat;
         let pair = self.get_chain_pair(Direction::Outgoing)?;
         let claim_fees_sat = match req.sat_per_vbyte {
             Some(sat_per_vbyte) => ESTIMATED_BTC_CLAIM_TX_VSIZE * sat_per_vbyte as u64,
             None => pair.clone().fees.claim_estimate(),
         };
         let server_fees_sat = pair.fees.server();
 
-        let user_lockup_amount_sat_without_service_fee =
-            receiver_amount_sat + claim_fees_sat + server_fees_sat;
-        let boltz_fees_sat = pair.fees.boltz(user_lockup_amount_sat_without_service_fee);
-        let user_lockup_amount_sat = user_lockup_amount_sat_without_service_fee + boltz_fees_sat;
-        self.validate_user_lockup_amount_for_chain_pair(&pair, user_lockup_amount_sat)?;
-        let lockup_fees_sat = self.estimate_lockup_tx_fee(user_lockup_amount_sat).await?;
+        let drain_payer_amount_sat = balance_sat;
+        let drain_lockup_fees_sat = self.estimate_drain_tx_fee().await?;
+        let drain_user_lockup_amount_sat = drain_payer_amount_sat - drain_lockup_fees_sat;
+        let drain_boltz_fees_sat = pair.fees.boltz(drain_user_lockup_amount_sat);
+        let drain_total_fees_sat =
+            drain_boltz_fees_sat + drain_lockup_fees_sat + claim_fees_sat + server_fees_sat;
+        let drain_receiver_amount_sat = drain_payer_amount_sat - drain_total_fees_sat;
```

Comment:
Perhaps it is better to calculate all the drain related variables inside the is_drain branch bellow?

Reply:
Agree, that's how I started, but then the question arose
> what if the user chooses `PayOnchainAmount::Receiver` of the exact receiver amount that would have been calculated by `drain` as well?

Then, that would be equivalent to choosing `PayOnchainAmount::Drain`.

If so, it seems natural to do `drain` instead of throwing an error.

To do that, we have to know what the amounts would have been in case of `drain`, which is why they're calculated before the `is_drain` branch below.

If instead we throw an error, let me know and this can definitely be simplified.

Reply:
> Then, that would be equivalent to choosing `PayOnchainAmount::Drain`.

No, we agreed that this would throw an error. The drain tx is only created if the user asks explicitly for drain, that was the whole point of the flag.

Reply:
Done: 62751b8448ff2f2a2c147e3bb43b1c709d7dc924